test/transforms.test.js

```diff
@@ -9,94 +9,148 @@ import {
+      });
+
+      expect(prompts[3]).toEqual({
+        type: 'confirm',
+        name: 'doTransform',
         message: 'How about this one?',
-        default: 'foo/**/*.bar.js',
-        transform: () => {}
-      }]
-    }]);
-
-    mockPackageJSON({
-      devDependencies: {
-        '@percy/sdk-test': '^1.0.0'
-      }
-    });
+        default: true
+      });
 
-    prompts = mockPrompts({
-      isSDK: true,
-      doTransform: false
+      expect(logger.stderr).toEqual([]);
+      expect(logger.stdout).toEqual([
+        '[percy] Migration complete!\n'
+      ]);
     });
-  });
 
-  it('confirms any SDK transforms', async () => {
-    await Migrate('@percy/sdk-test', '--skip-cli');
+    it('asks for filepaths to transform when confirmed', async () => {
+      prompts = mockPrompts({
+        isSDK: true,
+        doTransform: true,
+        filePaths: q => q.filter(q.default)
+          .then(f => f.sort())
+      });
 
-    expect(prompts[2]).toEqual({
-      type: 'confirm',
-      name: 'doTransform',
-      message: 'Run this transform?',
-      default: true
-    });
+      await Migrate('@percy/sdk-test', '--skip-cli');
 
-    expect(prompts[3]).toEqual({
-      type: 'confirm',
-      name: 'doTransform',
-      message: 'How about this one?',
-      default: true
-    });
+      expect(prompts[3]).toEqual({
+        type: 'input',
+        name: 'filePaths',
+        message: 'Which files?',
+        default: 'test/**/*.test.js',
+        filter: expect.any(Function)
+      });
 
-    expect(logger.stderr).toEqual([]);
-    expect(logger.stdout).toEqual([
-      '[percy] Migration complete!\n'
-    ]);
-  });
+      expect(prompts[5]).toEqual({
+        type: 'input',
+        name: 'filePaths',
+        message: 'Which files?',
+        default: 'test/**/*.test.js',
+        filter: expect.any(Function)
+      });
 
-  it('asks for filepaths to transform when confirmed', async () => {
-    prompts = mockPrompts({
-      isSDK: true,
-      doTransform: true,
-      filePaths: q => q.filter(q.default)
-        .then(f => f.sort())
+      expect(transformed).toEqual(
+        await globby('test/**/*.test.js')
+          .then(f => f.sort())
+      );
+
+      expect(logger.stderr).toEqual([]);
+      expect(logger.stdout).toEqual([
+        '[percy] Migration complete!\n'
+      ]);
     });
+  });
 
-    await Migrate('@percy/sdk-test', '--skip-cli');
+  describe('without matching paths', () => {
+    let transformed;
 
-    expect(prompts[3]).toEqual({
-      type: 'input',
-      name: 'filePaths',
-      message: 'Which files?',
-      default: 'test/**/*.test.js',
-      filter: expect.any(Function)
-    });
+    beforeEach(() => {
+      transformed = [];
+
+      mockMigrations([{
+        name: '@percy/sdk-test',
+        version: '^2.0.0',
+        transforms: [{
+          message: 'Never match',
+          default: 'no/files/should/match.ha',
+          transform: () => {}
+        }]
+      }]);
```

Comment:
Let's remove the variable reset, or make it `null`/`false`, then we can have the transform function set it to a value like `true`. That way the test can assert that this transform is never called by asserting on the initial value or asserting that it isn't the "transformed" value.

Reply:
```suggestion
      transformed = false;

      mockMigrations([{
        name: '@percy/sdk-test',
        version: '^2.0.0',
        transforms: [{
          message: 'Never match',
          default: 'no/files/should/match.ha',
          transform: () => (transformed = true)
        }]
      }]);
```